tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Process/ProcessStartCommon.cs

```diff
@@ -5,33 +5,42 @@
 
 using System;
 using System.Collections.Generic;
-using System.Collections.Specialized;
+using System.Collections.ObjectModel;
 using System.Diagnostics;
+using System.Text;
+using Datadog.Trace.AppSec;
 using Datadog.Trace.Configuration;
+using Datadog.Trace.DuckTyping;
 using Datadog.Trace.Logging;
 using Datadog.Trace.Tagging;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
 
 namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Process
 {
     internal static class ProcessStartCommon
     {
-        internal const IntegrationId IntegrationId = Configuration.IntegrationId.Process;
+        private const IntegrationId IntegrationId = Configuration.IntegrationId.Process;
+        private const string OperationName = "command_execution";
+        private const string ServiceName = "command";
+        private const int MaxCommandLineLength = 4096;
+
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ProcessStartCommon));
-        internal const string OperationName = "command_execution";
-        internal const string ServiceName = "command";
-        internal const int MaxCommandLineLength = 4096;
 
         internal static Scope CreateScope(ProcessStartInfo info)
         {
             if (info != null)
             {
-                return CreateScope(info.FileName, info.UseShellExecute ? null : info.Environment);
+#if NETCOREAPP3_1_OR_GREATER
+                return CreateScope(info.FileName, info.UseShellExecute ? null : info.Environment, info.UseShellExecute, info.Arguments, info.ArgumentList);
+#else
+                return CreateScope(info.FileName, info.UseShellExecute ? null : info.Environment, info.UseShellExecute, info.Arguments);
+#endif
             }
 
             return null;
```

Comment:
maybe this file could be nullable enable here

Reply:
Adding nullable like that?

```diff
+ #nullable enable
namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Process
{
    internal static class ProcessStartCommon
    {
```

Reply:
yes it allows to have method signatures that indicate whether we expect the object to be null, and callers are aware of it

`internal static Scope? CreateScope(ProcessStartInfo info)`